### ISSUE TITLE
feature request: better formatting around spacing and punctuation

### ISSUE THREAD

**Issue by jayarmstrong** [AnySoftKeyboard/AnySoftKeyboard#2503]
### Steps to reproduce
1. type a complete word
2. automatic space is added by AnySoft
3. user adds a comma

### Expected behaviour
Automatic space is removed to leave a grammatically correct sentence. 

### Actual behaviour
Automatic space remains, and sentence has a complete word, then a space, then the punctuation.
Example:
`Should there be spaces here , or not ?`

The above is just one example -- maybe more of a bug -- but it would be smart to have ASK correct the spacing for me while I'm typing, but allow manual editing. This is how Google's keyboard behaves, for example.

Android 10
ASK v1.10.1096

**Comment by menny** [comment]
I need a bit more information (also, this may have already been fixed with the latest beta).
When you type a word, do you pick it from the suggestions strip? in other words, what was the cause for the automatic space? (usually, that happens if you pick a word from the suggestions strip)

**Comment by tanitani-66** [comment]
The described problem doesn't happen on my Note 9. The punctuation marks remove the space before themselves. The auto spacing also happens when you swipe the words. 
It probably doesn't belong here yet I mention it: the opening parenthesis doesn't remove the space after itself.
Thanks, 
Gábor

**Comment by jayarmstrong** [comment]
@menny Yes, both when tapping the suggestion strip and when swipe/gesturing complete words, then adding punctuation. Sorry I wasn't more clear.

@tanitani-66 Thanks. Which version of ASK and Android are you using?

**Comment by tanitani-66** [comment]
ASK version 1.10-r3
on Android v10

**Comment by Motherboard** [comment]
Using Android 11, ASK version 1.10.1096 (gestures beta feature enabled) - I don't experience the problem described above. comma would be written right after the word, and a space would be added after it when selecting a word from the suggestion strip and typing a comma after it.

**Comment by jayarmstrong** [comment]
I just updated to 1.10.1109 and it's no longer adding the extra spaces. Thanks!

Since I'm the only one who reported this, I'll close it.

**Comment by jayarmstrong** [comment]
After further testing on the new version, I am seeing the issue when selecting a word from the autocomplete row and then adding an exclamation mark. ASK works well with other common punctuation.

**Comment by ballo** [comment]
Not only does the open parentheses fail to work correctly, so does the closed parentheses as it fails to be placed like a period if another form of punctuation is already there. Example:
Words (words words? )words
